### PR TITLE
oast: Fixes zaproxy/zaproxy#6997 Interactsh: multi threading issue

### PR DESCRIPTION
**Pull request by denniskniep**
Fixed Interactsh multi threading issue during register and deregister

Signed-off-by: Dennis Kniep <kniepdennis@gmail.com>

Fixes zaproxy/zaproxy#6997.

### CODE REVIEW

**Comment by thc202** [comment]
The corresponding issue should be linked: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by denniskniep** [comment]
Rebased

File: addOns/oast/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 ### Fixed
+- Fixed Interactsh multi threading issue during register and deregister
```
Comment (thc202) [inline addOns/oast/CHANGELOG.md:10]: ` (Issue 6997).`

**Comment by thc202** [comment]
I'd expect `getInteractions()` to be synchronized as well.

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
Looks good but has a conflict now.

**Comment by denniskniep** [comment]
rebased

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**